### PR TITLE
MGDOBR-1215: Manager: Remove manager-core test-jar usage

### PR DESCRIPTION
**Pull request by manstis**
[https://issues.redhat.com/browse/MGDOBR-1215](https://issues.redhat.com/browse/MGDOBR-1215)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: manager-parent/manager-v1/src/main/resources/db/migration/0.1/V0.1.5__create_shards.sql [hunk 1]
```diff
@@ -1,5 +1,3 @@
-DROP TABLE IF EXISTS SHARD;
```
Comment (danielezonca) [inline manager-parent/manager-v1/src/main/resources/db/migration/0.1/V0.1.5__create_shards.sql:1]: Is this related with the PR? Btw I agree that it seems strange to drop it everytime

File: manager-parent/manager-v1/src/test/resources/application.properties [hunk 2]
```diff
@@ -25,10 +25,20 @@ event-bridge.kafka.client.secret=client-secret
 event-bridge.kafka.security.protocol=PLAINTEXT
 event-bridge.kafka.sasl.mechanism=PLAIN
 
+# RHOAS Client
+event-bridge.rhoas.instance-api.host=http://test-instance-api.redhat.com
+event-bridge.rhoas.mgmt-api.host=http://test-management-api.redhat.com
+# RHOAS Client is unused by -core so Quarkus removes it from Arc unless we flag it as unremovable.
+quarkus.arc.unremovable-types=com.redhat.service.smartevents.rhoas.RhoasClient
```
Comment (danielezonca) [inline manager-parent/manager-v1/src/test/resources/application.properties:32]: I understand the comment but not the reason: if this is not used in `-core` why do we need to force to preserve it?

**Review by danielezonca — APPROVED** [review]
A couple of minor comments but LGTM 👍

File: manager-parent/manager-v1/src/main/resources/db/migration/0.1/V0.1.5__create_shards.sql (hunk 1 above)
Comment (manstis) [reply manager-parent/manager-v1/src/main/resources/db/migration/0.1/V0.1.5__create_shards.sql:1]: I added the `DROP...` to the DDL when I split `manager` into sub-modules as I also create the table in the tests for `manager-core`. Since `manager-v1` *was* including `manager-core` test-jar I had to drop the table otherwise tests failed (as the table had already been created). However the change to the existing DDL file led to Flyway to fail. I have therefore referred the DDL and removed the use of test-jar.

File: manager-parent/manager-v1/src/test/resources/application.properties (hunk 2 above)
Comment (manstis) [reply manager-parent/manager-v1/src/test/resources/application.properties:32]: Well... I can only assume Quarkus was pruning it as there is no direct usage. Without this setting for tests I get a warning that the class has been pruned and subsequently an unresolved type error. I can investigate more when I'm back at my laptop tomorrow.

**Review by r00ta — APPROVED** [review]
